### PR TITLE
Fix ArgumentNullException in AddNewtonsoftJson

### PR DESCRIPTION
**Pull request by sungam3r**
System.ArgumentNullException: 'Value cannot be null. '

```sh
GraphQL.NewtonsoftJson.dll!GraphQL.NewtonsoftJson.DocumentWriter.DocumentWriter(System.Action<Newtonsoft.Json.JsonSerializerSettings> configureSerializerSettings, GraphQL.Execution.IErrorInfoProvider errorInfoProvider)
GraphQL.Server.Transports.AspNetCore.NewtonsoftJson.dll!GraphQL.Server.GraphQLBuilderExtensions.AddNewtonsoftJson.AnonymousMethod__1(System.IServiceProvider p) 
```

### CODE REVIEW

File: src/Transports.AspNetCore.SystemTextJson/GraphQLBuilderExtensions.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static class GraphQLBuilderExtensions
             Action<JsonSerializerOptions> configureDeserializerSettings = null,
             Action<JsonSerializerOptions> configureSerializerSettings = null)
         {
-            builder.Services.AddSingleton<IGraphQLRequestDeserializer>(p => new GraphQLRequestDeserializer(configureDeserializerSettings));
+            builder.Services.AddSingleton<IGraphQLRequestDeserializer>(p => new GraphQLRequestDeserializer(configureDeserializerSettings ?? (_ => { })));
```
Comment (sungam3r) [inline src/Transports.AspNetCore.SystemTextJson/GraphQLBuilderExtensions.cs:32]: It was not necessary, but I unified the behavior with another serializer.

File: src/Transports.AspNetCore.NewtonsoftJson/GraphQLBuilderExtensions.cs [hunk 2]
```diff
@@ -29,8 +29,8 @@ public static class GraphQLBuilderExtensions
             Action<JsonSerializerSettings> configureDeserializerSettings = null,
             Action<JsonSerializerSettings> configureSerializerSettings = null)
         {
-            builder.Services.AddSingleton<IGraphQLRequestDeserializer>(p => new GraphQLRequestDeserializer(configureDeserializerSettings));
-            builder.Services.Replace(ServiceDescriptor.Singleton<IDocumentWriter>(p => new DocumentWriter(configureSerializerSettings, p.GetService<IErrorInfoProvider>() ?? new ErrorInfoProvider())));
+            builder.Services.AddSingleton<IGraphQLRequestDeserializer>(p => new GraphQLRequestDeserializer(configureDeserializerSettings ?? (_ => { })));
+            builder.Services.Replace(ServiceDescriptor.Singleton<IDocumentWriter>(p => new DocumentWriter(configureSerializerSettings ?? (_ => { }), p.GetService<IErrorInfoProvider>() ?? new ErrorInfoProvider())));
```
Comment (sungam3r) [inline src/Transports.AspNetCore.NewtonsoftJson/GraphQLBuilderExtensions.cs:33]:
```c#
 /// <summary>
        /// Initializes a new instance of the <see cref="DocumentWriter"/> class with the specified settings.
        /// Configuration defaults to no indenting and the specified instance of the <see cref="ErrorInfoProvider"/> class.
        /// </summary>
        /// <param name="configureSerializerSettings">Specifies a callback used to configure the JSON serializer</param>
        /// <param name="errorInfoProvider">Specifies the <see cref="IErrorInfoProvider"/> instance to use to serialize GraphQL errors</param>
        public DocumentWriter(Action<JsonSerializerSettings> configureSerializerSettings, IErrorInfoProvider errorInfoProvider)
            : this(BuildSerializer(false,
                configureSerializerSettings ?? throw new ArgumentNullException(nameof(configureSerializerSettings)),
                errorInfoProvider ?? throw new ArgumentNullException(nameof(errorInfoProvider))))
        {
        }
```

It is obvious that we stepped on our own exception here.

**Comment by sungam3r** [comment]
I will release 4.2.1 ASAP.

**Comment by Shane32** [comment]
👍

**Pull request merged**